Added test case for using RequestDumpingHandler

jstourac opened the pull request:
Added test case for using RequestDumpingHandler into integration tests; test implementation for RFE EAP7-428

jstourac committed:
Added test case for using RequestDumpingHandler

Added test case for using RequestDumpingHandler into integration tests; RFE TestImpl JBQA-11996

ctomc commented on testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/RequestDumpingHandlerTestCase.java:
```diff
@@ -0,0 +1,299 @@
+            .append(PERIODIC_ROTATING_FILE_HANDLER, "FILE");
+
+    @Before
+    public void setup() throws Exception {
```
it would be cleaner if you would use ServerSetupTask for this kind of things. most of tests use it.
you just add annotation @ServerSetup(name-of-your-setup-class.class)
and you get access to mgmt client and friends.

ctomc commented on testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/RequestDumpingHandlerTestCase.java:
```diff
@@ -0,0 +1,299 @@
+        ModelNode file = new ModelNode();
+        file.get("relative-to").set(relativeTo);
+        file.get("path").set(logFilePrefix + LOG_FILE_SUFFIX);
+        ModelNode op = Util.getWriteAttributeOperation(aLogAddr, "file", file);
```
This feels very volatile, as it changes log location for whole server. 
I think it would be better if test would add new logger (io.undertow.request.dump) and append only to new file handler that would than have only request logs in it.
Also the logic for looking if content is in file would be much easier.

ctomc commented:
this is ok to test

jstourac committed:
used ServerSetupTask class and separate logger

- moved server setting into separate ServerSetupTask class
- used separate logger for dumping messages instead of changing global server logfile

jstourac commented:
Thanks Tomaz for your review!

I have rewritten test according to your proposes - used ServerSetupTask for server configuration preparation and also used separate logger for dumping messages.

I am not sure why this test failed for wildfly-ci as it is working for me on my local machine... lets try again...

jstourac commented:
test fails seems unrelated...

bstansberry commented:
retest this please

jstourac committed:
Added wait loop when searching in log file, added new line when reading file content

- Added wait loop with 3s timeout when searching for basic content in the log file as on highly
  loaded machines there might be some delay before all data is written down on the HDD.
- Added new line when reading file content to respect real log file format. This change although
  has no impact on test itself - just on INFO message that is printed to expose what has been
  actually read from the log file.

jstourac commented:
Thanks for retesting!

I am no sure why exactly this test: RequestDumpingHandlerTestCase.testReqDumpHandlerOn has failed as on my machine this test and even ALL others are successfully passing.

I added sleep loop when searching for data in log file as I am suspecting that there was not enough time for system to write data down into log file which is examinate. Next test run will show...

jstourac committed:
Fix: quoted variable values used as patterns

- fixed: now quoting variable values used as patterns for searching in
  log file.

jstourac commented:
Failed tests seems unrelated, retest this please.

jstourac commented:
Failure seems unrelated...

jstourac committed:
Added two tests: dump HTTPS request and request with upgrade to Websocket protocol

Added basic check of request dump values of request via HTTPS and HTTP with upgrade
to websocket protocol. This is just to ensure that dumping values are OK when used secured protocol.

To preserve readibility code is split in two files - in *TestCase there is server setup and tests definition
whereas in *TestImpl there is particular implementation of what is tested.

jstourac commented:
Added test checking basic dumped values in log when request from HTTPS is performed and test when http upgrade to websockets is performed. Code is now splitted into two files to preserve readibility.

Could some of admins review it please?

jstourac commented:
Thanks for review and merge Jason!
